Sync/Synchronizers/Bookmarks/BookmarksSynchronizer.swift

```diff
@@ -197,7 +197,10 @@ open class BufferingBookmarksSynchronizer: TimestampedSingleCollectionSynchroniz
                     log.debug("Validating completed buffer download.")
                     return buffer.validate().bind { validationResult in
                         if let invalidError = validationResult.failureValue as? BufferInvalidError {
-                            self.statsSession.validationStats = self.validationStatsFrom(error: invalidError)
+                            return buffer.getUpstreamRecordCount() >>== { checked in
```

Comment:
If this fails — e.g., because the database is now closed — we won't include validation stats in the result at all.

I suggest you bind here rather than chaining, and do something sensible if we can't retrieve the count (e.g., send a sentinel value, or omit the count).